File: 2025-09-25 - ListaPrioridadePQ/Exemplo_C#/ListaPrioridadeInterface.cs
// ------------------------------------------------------------------------
// ListaPrioridadeInterface.cs
// Interface que define os métodos da lista de prioridade
// ------------------------------------------------------------------------
public interface ListaPrioridadeInterface
{
    bool inserir(Node novoNode);                  // Inserir mantendo a prioridade
    bool remover(string codigoDisciplina);        // Remover por código da disciplina
    Node buscar(string codigoDisciplina);         // Buscar por código
    bool atualizar(Node nodeAtualizado);          // Atualizar disciplina existente
    void imprimirLista();                         // Imprimir todos os elementos
    int tamanho();                                // Retorna o tamanho da lista
}

File: 2025-09-25 - Lista Prioridade/Exemplo_Java/ListaPrioridadeInterface.java
// ------------------------------------------------------------------------
// ListaPrioridadeInterface.java
// Interface que define os métodos públicos da lista de prioridade
// ------------------------------------------------------------------------
public interface ListaPrioridadeInterface {
   boolean inserir(Node novoNode);      // Inserir mantendo a prioridade
   boolean remover(String codigoDisciplina); // Remover por código da disciplina
   Node buscar(String codigoDisciplina);    // Buscar por código
   boolean atualizar(Node nodeAtualizado);  // Atualizar disciplina existente
   void imprimirLista();                   // Imprimir todos os elementos
   int tamanho();                          // Retorna o tamanho da lista
}
